src/self_driving/planning/pilot.cpp

```diff
@@ -71,6 +71,10 @@ void Pilot::PerformPlanning() {
     return;
   }
 
+  // Compute memory information used to satisfy the memory constraint
+  PilotUtil::ComputeTableSizeRatios(forecast_.get(), task_manager_, txn_manager_, catalog_, &memory_info_);
+  PilotUtil::ComputeTableIndexSizes(txn_manager_, catalog_, &memory_info_);
```

Comment:
Why do these take the `TransactionManager` rather than a single `TransactionContext`?

Note:  I think these should probably be merged into one function that does take the `TransactionManager`.  This would allow you to query the catalog (i.e., `pg_stats` and the indexes) inside of the same transaction that you resolved the OIDs (avoids TOCTOU concerns from concurrent DDL).